File: src/Expenda.Domain/Entities/Expense.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Expenda.Domain.Entities.Base;

namespace Expenda.Domain.Entities;

public class Expense : AuditableBaseEntity
{
    [Required]
    [MinLength(1)]
    [MaxLength(250)]
    public string Name { get; set; } = null!;
    
    [MaxLength(1000)]
    public string? Description { get; set; }
    
    [Required]
    public double Price { get; set; }
    
    [Required]
    public int Quantity { get; set; }

    [Required]
    public DateTime TransactionDate { get; set; }
    
    [Required]
    [ForeignKey(nameof(ApplicationUser))]
    public int OwnerId { get; set; }
}

File: src/Expenda.Mobile/lib/models/expense.dart
class ExpenseRequest {
  final String name;
  final String description;
  final int quantity;
  final double price;
  final DateTime transactionDate;

  ExpenseRequest(this.name, this.description, this.quantity, this.price,
      this.transactionDate);
}

class ExpenseResponse extends ExpenseRequest {
  final int id;

  ExpenseResponse(this.id, String name, String description, int quantity,
      double price, DateTime transactionDate)
      : super(name, description, quantity, price, transactionDate);

  factory ExpenseResponse.fromJson(Map<String, dynamic> json) {
    return ExpenseResponse(json['id'], json['name'], json['description'],
        json['quantity'], json['price'], json['transaction_date']);
  }
}
